pyo3-bytes/src/bytes.rs

```diff
@@ -62,6 +63,66 @@ impl PyBytes {
     pub fn as_slice(&self) -> &[u8] {
         self.as_ref()
     }
+
+    /// Slice the underlying buffer using a Python slice object
+    ///
+    /// This should behave the same as Python's byte slicing:
+    ///     - `ValueError` if step is zero
+    ///     - Negative indices a-ok
+    ///     - If start/stop are out of bounds, they are clipped to the bounds of the buffer
+    ///     - If start > stop, the slice is empty
+    ///
+    /// This is NOT exposed to Python under the `#[pymethods]` impl
+    fn slice(&self, slice: &Bound<'_, PySlice>) -> PyResult<PyBytes> {
+        let len_isize = self.0.len() as isize;
```

Comment:
Similarly, we can see the types with rust-analyzer, and we don't need to have the type be part of the variable name. So just name this like `bytes_length`.

Reply:
True that. Old habit!